File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int solution(string s) {
    string answer = "", tmp = "";
    vector<string> num = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    
    for(int i = 0; i < s.length(); i++) {
        if('0' <= s[i] && s[i] <= '9') answer += s[i];
        else tmp += s[i];
        
        if(find(num.begin(), num.end(), tmp) != num.end()) {
            answer += to_string(find(num.begin(), num.end(), tmp) - num.begin());
            tmp = ""; // tmp를 다시 빈문자열로 바꿈
        }
    }
    return stoi(answer);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def convert(s):
    if s == "zero":
        return '0'
    elif s == "one":
        return '1'
    elif s == "two":
        return '2'
    elif s == "three":
        return '3'
    elif s == "four":
        return '4'
    elif s == "five":
        return '5'
    elif s == "six":
        return '6'
    elif s == "seven":
        return '7'
    elif s == "eight":
        return '8'
    elif s == "nine":
        return '9'
    
def solution(s):
    answer = ""
    tmp = ""
    numbers = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    
    for c in s:
        if c >= '0' and c <= '9':
            answer += c
        else:
            tmp += c
            if tmp in numbers:
                answer += convert(tmp)
                tmp = ""               
            
    return int(answer)